src/platform/i386/chal_pgtbl.c

```diff
@@ -0,0 +1,188 @@
+#include "shared/cos_types.h"
+#include "captbl.h"
+#include "pgtbl.h"
+#include "cap_ops.h"
+#include "liveness_tbl.h"
+#include "retype_tbl.h"
+
+int
+chal_pgtbl_kmem_act(pgtbl_t pt, u32_t addr, unsigned long *kern_addr, unsigned long **pte_ret)
```

Comment:
Doesn't the typedef for `pgtbl_t` need to be in the architecture-specific code as well?

Reply:
Yes, they will be moved to chal soon